lib/job.jsonnet

```diff
@@ -84,11 +84,9 @@ function(jobName, agentEnv={}, stepEnvFile='', patchFunc=identity) patchFunc({
   local stepEnv =
     [
       {
-        local trim(v) = if std.startsWith(v, '"') && std.endsWith(v, '"')
-          then std.substr(v, 1, std.length(v) - 2) else v,
         local kv = std.splitLimit(l, '=', 1),
         name: kv[0],
-        value: trim(kv[1]),
+        value: std.parseJson(kv[1]),
```

Comment:
What would happen if the value is malformed? Would that fail?

Reply:
Malformed how? Note at this point buildkite has already accepted the env so there's some validation that has already happened

Reply:
I don't know about specific use cases, but it seems risky to try to parse every env variable as JSON. For example: "{\"baz\": \[\"a\", \"b\",\]}", this should fail because of the second comma.

The values might be enclosed in quotes, but the content is not necessarily JSON. This should work fine for simple values like numbers, but what if the string is more complicated?

Reply:
Ah got it, this invalid json appears to be passed through as a string in the case of failure.